tests/core/test_hook_save_scalar.py

```diff
@@ -343,7 +343,8 @@ def check_trials(out_dir, save_steps, saved_scalars=None):
             if eval_steps:  # need this check for bias and gradients
                 assert len(set(save_steps["EVAL"]) & set(eval_steps)) == len(save_steps["EVAL"])
     scalar_list = trial.tensor_names(regex="^scalar")
-    assert len(set(saved_scalars) & set(scalar_list)) == len(saved_scalars)
+    if saved_scalars:
+        assert len(set(saved_scalars) & set(scalar_list)) == len(saved_scalars)
```

Comment:
why this special intersection ? 
Shouldn't len(Saved_scalars) == len(scalar_list) 

What is the reason where one list can be more than other list?